### PR TITLE
Add table googledirectory_role_assignment. Closes #7

### LINKED ISSUE #7: Add table googledirectory_role_assignment
**References**
https://developers.google.com/admin-sdk/directory/reference/rest/v1/roleAssignments
https://registry.terraform.io/providers/hashicorp/googleworkspace/latest/docs

### PR DESCRIPTION
**Pull request by bigdatasourav**
# Example query results
<details>
  <summary>Results</summary>
  
```
> select
  role_assignment_id,
  role_id,
  assigned_to,
  scope_type
from
  googledirectory_role_assignment;
+--------------------+-------------------+-----------------------+------------+
| role_assignment_id | role_id           | assigned_to           | scope_type |
+--------------------+-------------------+-----------------------+------------+
| 52236313256001537  | 52236313256001537 | 100347617708023627093 | CUSTOMER   |
| 52236313256001540  | 52236313256001546 | 116941638922292931139 | CUSTOMER   |
| 52236313256001538  | 52236313256001537 | 115446550006107555454 | CUSTOMER   |
| 52236313256001541  | 52236313256001537 | 116941638922292931139 | CUSTOMER   |
+--------------------+-------------------+-----------------------+------------+
> select
  role_assignment_id,
  role_id,
  assigned_to,
  scope_type
from
  googledirectory_role_assignment
where
  role_id = '52236313256001537';
+--------------------+-------------------+-----------------------+------------+
| role_assignment_id | role_id           | assigned_to           | scope_type |
+--------------------+-------------------+-----------------------+------------+
| 52236313256001538  | 52236313256001537 | 115446550006107555454 | CUSTOMER   |
| 52236313256001537  | 52236313256001537 | 100347617708023627093 | CUSTOMER   |
| 52236313256001541  | 52236313256001537 | 116941638922292931139 | CUSTOMER   |
+--------------------+-------------------+-----------------------+------------+

> select * from googledirectory_role_assignment where user_key = '115446550006107555454'
+--------------------+-------------------+-----------------------+-------------+-----------------------+---------------------------------------------------------------------------+------------------------
| role_assignment_id | role_id           | assigned_to           | customer_id | user_key              | etag                                                                      | kind                   
+--------------------+-------------------+-----------------------+-------------+-----------------------+---------------------------------------------------------------------------+------------------------
| 52236313256001538  | 52236313256001537 | 115446550006107555454 | <null>      | 115446550006107555454 | "jU41SRtWIAfoLzcTxu8ldh8oAWnTyj8bXpPMVPE5dps/eS-0h_Ik6-qm9poXkeraHJFdP3I" | admin#directory#roleAss
+--------------------+-------------------+-----------------------+-------------+-----------------------+---------------------------------------------------------------------------+------------------------
> select * from googledirectory.googledirectory_role_assignment where role_id ='52236313256001537'
+--------------------+-------------------+-----------------------+-------------+----------+---------------------------------------------------------------------------+--------------------------------+----
| role_assignment_id | role_id           | assigned_to           | customer_id | user_key | etag                                                                      | kind                           | org
+--------------------+-------------------+-----------------------+-------------+----------+---------------------------------------------------------------------------+--------------------------------+----
| 52236313256001537  | 52236313256001537 | 100347617708023627093 | <null>      | <null>   | "jU41SRtWIAfoLzcTxu8ldh8oAWnTyj8bXpPMVPE5dps/PMMvoTnRkFzAHRptlDOmEfTqgKA" | admin#directory#roleAssignment | <nu
| 52236313256001538  | 52236313256001537 | 115446550006107555454 | <null>      | <null>   | "jU41SRtWIAfoLzcTxu8ldh8oAWnTyj8bXpPMVPE5dps/eS-0h_Ik6-qm9poXkeraHJFdP3I" | admin#directory#roleAssignment | <nu
| 52236313256001541  | 52236313256001537 | 116941638922292931139 | <null>      | <null>   | "jU41SRtWIAfoLzcTxu8ldh8oAWnTyj8bXpPMVPE5dps/GTxm1y34UOmGNJGRsQ7nTI8x3TM" | admin#directory#roleAssignment | <nu
+--------------------+-------------------+-----------------------+-------------+----------+---------------------------------------------------------------------------+--------------------------------+----
> select * from googledirectory.googledirectory_role_assignment
+--------------------+-------------------+-----------------------+-------------+----------+---------------------------------------------------------------------------+--------------------------------+----
| role_assignment_id | role_id           | assigned_to           | customer_id | user_key | etag                                                                      | kind                           | org
+--------------------+-------------------+-----------------------+-------------+----------+---------------------------------------------------------------------------+--------------------------------+----
| 52236313256001540  | 52236313256001546 | 116941638922292931139 | <null>      | <null>   | "jU41SRtWIAfoLzcTxu8ldh8oAWnTyj8bXpPMVPE5dps/dSi_tD8Wk6kNlnL4LoAdQR_OVrg" | admin#directory#roleAssignment | <nu
| 52236313256001541  | 52236313256001537 | 116941638922292931139 | <null>      | <null>   | "jU41SRtWIAfoLzcTxu8ldh8oAWnTyj8bXpPMVPE5dps/GTxm1y34UOmGNJGRsQ7nTI8x3TM" | admin#directory#roleAssignment | <nu
| 52236313256001537  | 52236313256001537 | 100347617708023627093 | <null>      | <null>   | "jU41SRtWIAfoLzcTxu8ldh8oAWnTyj8bXpPMVPE5dps/PMMvoTnRkFzAHRptlDOmEfTqgKA" | admin#directory#roleAssignment | <nu
| 52236313256001538  | 52236313256001537 | 115446550006107555454 | <null>      | <null>   | "jU41SRtWIAfoLzcTxu8ldh8oAWnTyj8bXpPMVPE5dps/eS-0h_Ik6-qm9poXkeraHJFdP3I" | admin#directory#roleAssignment | <nu
+--------------------+-------------------+-----------------------+-------------+----------+---------------------------------------------------------------------------+--------------------------------+----

```
</details>

### CODE REVIEW

File: googledirectory/table_googledirectory_role_assignment.go [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+		return nil, err
+	}
+
+	// Set default value to my_customer, to represent current account
+	customerID := "my_customer"
+	if d.KeyColumnQuals["customer_id"] != nil {
+		customerID = d.KeyColumnQuals["customer_id"].GetStringValue()
+	}
+
+	resp := service.RoleAssignments.List(customerID)
```
Comment (Subhajit97) [inline googledirectory/table_googledirectory_role_assignment.go:93]: Add support to filter using `roleId` and `userKey` also.
https://pkg.go.dev/google.golang.org/api@v0.52.0/admin/directory/v1#RoleAssignmentsListCall.RoleId
https://pkg.go.dev/google.golang.org/api@v0.52.0/admin/directory/v1#RoleAssignmentsListCall.UserKey

File: docs/tables/googledirectory_role_assignment.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+select
+  role_assignment_id,
+  role_id,
+  assigned_to,
+  scope_type
+from
+  googledirectory_role_assignment;
+```
+
+### Get role assigments by role ID
```
Comment (Subhajit97) [inline docs/tables/googledirectory_role_assignment.md:19]:
```suggestion
### Get role assignments by role ID
```

File: docs/tables/googledirectory_role_assignment.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+select
+  role_assignment_id,
+  role_id,
+  assigned_to,
+  scope_type
+from
+  googledirectory_role_assignment
+where
+  role_id = '522363132560015';
+```
```
Comment (Subhajit97) [inline docs/tables/googledirectory_role_assignment.md:31]: Can we add an example to list role assignments by user_id?

File: googledirectory/table_googledirectory_role_assignment.go [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+	if d.KeyColumnQuals["customer_id"] != nil {
+		customerID = d.KeyColumnQuals["customer_id"].GetStringValue()
+	}
+
+	var roleId string
+	if d.KeyColumnQuals["role_id"] != nil {
+		roleId = d.KeyColumnQuals["role_id"].GetStringValue()
+	}
+
+	resp := service.RoleAssignments.List(customerID).RoleId(roleId)
```
Comment (Subhajit97) [inline googledirectory/table_googledirectory_role_assignment.go:112]: handle `notFound` case

File: googledirectory/table_googledirectory_role_assignment.go [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+	if err := resp.Pages(ctx, func(page *admin.RoleAssignments) error {
+		for _, assignment := range page.Items {
+			d.StreamListItem(ctx, assignment)
+		}
+		return nil
+	}); err != nil {
+		return nil, err
+	}
+
+	return nil, err
```
Comment (Subhajit97) [inline googledirectory/table_googledirectory_role_assignment.go:125]:
```suggestion
	return nil, nil
```

**Review by Subhajit97 — APPROVED** [review]

**Review by rajlearner17 — APPROVED** [review]
LGTM

**Pull request merged**